src/serena/mcp.py

```diff
@@ -68,29 +65,55 @@ def execute_fn(ctx: Context, **kwargs) -> str:  # type: ignore
         parameters=parameters,
         fn_metadata=func_arg_metadata,
         is_async=is_async,
-        context_kwarg="ctx",
+        context_kwarg=None,
     )
 
 
 def create_mcp_server() -> FastMCP:
     argv = sys.argv[1:]
-    if len(argv) != 1:
-        print("\nUsage: mcp_server <.yml project file>", file=sys.stderr)
-        sys.exit(1)
 
-    project_file_path = argv[0]
-    agent = SerenaAgent(project_file_path)
+    if (len(argv) == 1 and argv[0] == "--help") or len(argv) > 1:
+        print("\nUsage: mcp_server [.yml project file]", file=sys.stderr)
+        sys.exit(0)
+
+    mcp: FastMCP | None = None
+
+    def update_tools() -> None:
+        """Update the tools in the MCP server."""
+        # Tools may change as a result of project activation.
```

Comment:
We should document this in the readme (or did I miss it)?

Reply:
Document what exactly?
I think what you're looking for is documented in `serena_config.template.yml`, because that's where the setting is controlled that affects this behaviour.
I don't want to duplicate it in the README.

Reply:
Ah, ok, there is enough